### PR TITLE
Was created two new test with BasePage

### PR DESCRIPTION
**Pull request by GiorgioJovanni**
https://tarsit.atlassian.net/jira/software/projects/TARS/boards/1?selectedIssue=TARS-21

### CODE REVIEW

File: Tests/Tests UI/test_register_page.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+from Tests.PageObject.RegisterPage import RegisterPage
+from Tests.PageObject.locators import BasePageLocators
+
+
+def test_guest_can_register(browser):
+    link = "http://parabank.parasoft.com"
```
Comment (AramaisTyshchenko) [inline Tests/Tests UI/test_register_page.py]: Please create constant for this link, like base url

File: Tests/Tests UI/test_register_page.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    page = RegisterPage(browser, link)
+    page.open()
+    page.click_on_button(*BasePageLocators.BUTTON_REGISTER)
+    page.guest_can_go_to_register_page()
+    page.register_new_fake_user()
+
+
+def test_guest_use_buttons_gohome_contact_aboutus_from_register_page(browser):
+    link = "http://parabank.parasoft.com"
```
Comment (AramaisTyshchenko) [inline Tests/Tests UI/test_register_page.py]: Add negative tests for the register page

**Review by AramaisTyshchenko — CHANGES_REQUESTED** [review]

File: Tests/PageObject/BasePage.py [hunk 3]
```diff
@@ -38,3 +41,39 @@ def open(self):
 
     def click_on_button(self, how, what):
         self.browser.find_element(how, what).click()
+
+    def button_go_home_is_working(self):
```
Comment (AramaisTyshchenko) [inline Tests/PageObject/BasePage.py]: please rename each button to just button_go_home, and so on

Comment (AramaisTyshchenko) [inline Tests/PageObject/BasePage.py]: please rename each button to just button_go_home, and so on

File: Tests/PageObject/RegisterPage.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+from Tests.PageObject.locators import RegisterPageLocators
+
+fake = faker.Faker()
+password = fake.password()
+first_name = fake.first_name()
+last_name = fake.last_name()
+
+
+class RegisterPage(BasePage):
+    def guest_can_go_to_register_page(self):
```
Comment (AramaisTyshchenko) [inline Tests/PageObject/RegisterPage.py]: Rename to Open_register_page

Comment (AramaisTyshchenko) [inline Tests/PageObject/RegisterPage.py]: Rename to Open_register_page

File: Tests/PageObject/RegisterPage.py [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        self.should_be_customer_form()
+
+    def should_be_customer_form(self):
+        assert self.is_element_present(*RegisterPageLocators.RIGHT_PANEL), "Right panel is not present"
```
Comment (AramaisTyshchenko) [inline Tests/PageObject/RegisterPage.py]: each element should be described like a separete function

Comment (AramaisTyshchenko) [inline Tests/PageObject/RegisterPage.py]: each element should be described like a separete function

Comment (AramaisTyshchenko) [reply Tests/PageObject/RegisterPage.py]: for this element you can use element.is_displayed() for verification

File: Tests/PageObject/RegisterPage.py [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        assert self.is_element_present(*RegisterPageLocators.CUSTOMER_FORM), "Customer form is not present"
+
+    def register_new_fake_user(self):
+        self.browser.find_element(*RegisterPageLocators.FIELD_FIRST_NAME).send_keys(first_name)
```
Comment (AramaisTyshchenko) [inline Tests/PageObject/RegisterPage.py]: same here, each element should be a separate function. register_new_fake_user(self) is a helper function and should be moved away from this class

Comment (AramaisTyshchenko) [inline Tests/PageObject/RegisterPage.py]: same here, each element should be a separata function. register_new_fake_user(self) is a helper function and should be moved away from this class

File: Tests/PageObject/RegisterPage.py [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+        self.browser.find_element(*RegisterPageLocators.FIELD_PHONE).send_keys('+01234241231')
+        self.browser.find_element(*RegisterPageLocators.FIELD_SNN).send_keys('23412341234123')
+        self.browser.find_element(*RegisterPageLocators.FIELD_USERNAME).send_keys(f"-1{first_name}1-")
+        self.browser.find_element(*RegisterPageLocators.FIELD_PASSWORD).send_keys(password)
+        self.browser.find_element(*RegisterPageLocators.FIELD_CONFIRM).send_keys(password)
+        self.click_on_button(*RegisterPageLocators.BUTTON_REGISTER)
+        assert self.is_not_element_present(*RegisterPageLocators.BUTTON_REGISTER), "Button register is not disappear"
+
+    def negative_register_new_fake_user(self):
+        self.browser.find_element(*RegisterPageLocators.FIELD_FIRST_NAME).send_keys(first_name)
```
Comment (AramaisTyshchenko) [inline Tests/PageObject/RegisterPage.py]: same here, and this function checks only one error. but there could be about 9 error messages

Comment (AramaisTyshchenko) [inline Tests/PageObject/RegisterPage.py]: same here, and this function checks only one error. but there could be about 9 error messages

File: Tests/Tests UI/test_base_page.py [hunk 8]
```diff
@@ -0,0 +1,44 @@
+from Tests.PageObject.BasePage import BasePage
+from Tests.PageObject.locators import BasePageLocators
+
+base_link = "http://parabank.parasoft.com"
```
Comment (AramaisTyshchenko) [inline Tests/Tests UI/test_base_page.py]: MOve this to separate python module where you store constants

Comment (AramaisTyshchenko) [inline Tests/Tests UI/test_base_page.py]: MOve this to separete python module where you store constants

File: Tests/Tests UI/test_base_page.py [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+    page.open()
+    page.click_on_button(*BasePageLocators.BUTTON_REGISTER)
+    page.button_go_contact_is_working()
+    page.open()
+    page.click_on_button(*BasePageLocators.BUTTON_REGISTER)
+    page.button_go_aboutus_is_working()
+
+
+class TestUpMenu:
+    def test_guest_use_button_about_us(self, browser):
```
Comment (AramaisTyshchenko) [inline Tests/Tests UI/test_base_page.py]: please add a fixture that would run before all tests and start the browser and execute commands

page = BasePage(browser, base_link)
        page.open()
        
So the browser would be loaded only once before all tests

Comment (AramaisTyshchenko) [inline Tests/Tests UI/test_base_page.py]: please add a fixture that would run before all tests and start the browser and execute commands

page = BasePage(browser, base_link)
        page.open()
        
So the browser would be loaded only once before all tests

File: Tests/Tests UI/test_register_page.py [hunk 10]
```diff
@@ -0,0 +1,34 @@
+from Tests.PageObject.RegisterPage import RegisterPage
+from Tests.PageObject.locators import BasePageLocators
+
+base_link = "http://parabank.parasoft.com"
+
+
```
Comment (AramaisTyshchenko) [inline Tests/Tests UI/test_register_page.py]: please add a fixture that would run before all tests and start the browser and execute commands

page = RegisterPage(browser, base_link)
    page.open()
        
So the browser would be loaded only once before all tests

Comment (AramaisTyshchenko) [inline Tests/Tests UI/test_register_page.py]: please add a fixture that would run before all tests and start the browser and execute commands

page = RegisterPage(browser, base_link)
    page.open()
        
So the browser would be loaded only once before all tests

File: Tests/Tests UI/test_register_page.py [hunk 11]
```diff
@@ -0,0 +1,34 @@
+from Tests.PageObject.RegisterPage import RegisterPage
+from Tests.PageObject.locators import BasePageLocators
+
+base_link = "http://parabank.parasoft.com"
```
Comment (AramaisTyshchenko) [inline Tests/Tests UI/test_register_page.py]: MOve this to separate python module where you store constants

Comment (AramaisTyshchenko) [inline Tests/Tests UI/test_register_page.py]: MOve this to separate python module where you store constants

File: Tests/Tests UI/test_register_page.py [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+
+
+def test_guest_can_register(browser):
+    page = RegisterPage(browser, base_link)
+    page.open()
+    page.click_on_button(*BasePageLocators.BUTTON_REGISTER)
```
Comment (AramaisTyshchenko) [inline Tests/Tests UI/test_register_page.py]: please describe button as a function for the register page class

Comment (AramaisTyshchenko) [inline Tests/Tests UI/test_register_page.py]: please describe button as a function for the register page class

File: Tests/Tests UI/test_register_page.py [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+    page.guest_can_go_to_register_page()
+    page.register_new_fake_user()
+
+
+def test_negative_guest_can_register(browser):
```
Comment (AramaisTyshchenko) [inline Tests/Tests UI/test_register_page.py]: this test covers only one error out of 9 errors

Comment (AramaisTyshchenko) [inline Tests/Tests UI/test_register_page.py]: this test covers only one error out of 9 errors

File: Tests/PageObject/RegisterPage.py [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+        self.browser.find_element(*RegisterPageLocators.FIELD_CITY).send_keys('New York')
+        self.browser.find_element(*RegisterPageLocators.FIELD_STATE).send_keys('New York')
+        self.browser.find_element(*RegisterPageLocators.FIELD_ZIPCODE).send_keys('0007')
+        self.browser.find_element(*RegisterPageLocators.FIELD_PHONE).send_keys('+01234241231')
+        self.browser.find_element(*RegisterPageLocators.FIELD_SNN).send_keys('23412341234123')
+        self.browser.find_element(*RegisterPageLocators.FIELD_USERNAME).send_keys(f"-1{first_name}1-")
+        self.browser.find_element(*RegisterPageLocators.FIELD_PASSWORD).send_keys(password)
+        self.browser.find_element(*RegisterPageLocators.FIELD_CONFIRM).send_keys(password)
+        self.click_on_button(*RegisterPageLocators.BUTTON_REGISTER)
+        assert self.is_not_element_present(*RegisterPageLocators.BUTTON_REGISTER), "Button register is not disappear"
```
Comment (AramaisTyshchenko) [inline Tests/PageObject/RegisterPage.py]: please verify that this text appeared "Your account was created successfully. You are now logged in.
to be sure that account was created

File: Tests/PageObject/RegisterPage.py (hunk 5 above)
Comment (GiorgioJovanni) [reply Tests/PageObject/RegisterPage.py]: done

File: Tests/PageObject/BasePage.py [hunk 15]
```diff
@@ -38,3 +41,42 @@ def open(self):
 
     def click_on_button(self, how, what):
         self.browser.find_element(how, what).click()
+
+    def button_go_home(self):
```
Comment (AramaisTyshchenko) [inline Tests/PageObject/BasePage.py]: create a separate Class

File: Tests/PageObject/RegisterPage.py [hunk 16]
```diff
@@ -0,0 +1,72 @@
...
+class RegisterPage(BasePage):
+
+    def should_be_panel(self):
+        assert self.is_element_present(*RegisterPageLocators.RIGHT_PANEL), "Right panel is not present"
+
+    def should_be_form(self):
+        assert self.is_element_present(*RegisterPageLocators.CUSTOMER_FORM), "Customer form is not present"
+
+    def add_first_name(self):
+        self.browser.find_element(*RegisterPageLocators.FIELD_FIRST_NAME).send_keys(first_name)
```
Comment (AramaisTyshchenko) [inline Tests/PageObject/RegisterPage.py]: should be separate operations, found and click. CLick in Test!

File: Tests/PageObject/RegisterPage.py [hunk 17]
```diff
@@ -0,0 +1,72 @@
...
+
+class RegisterPage(BasePage):
+
+    def should_be_panel(self):
+        assert self.is_element_present(*RegisterPageLocators.RIGHT_PANEL), "Right panel is not present"
+
+    def should_be_form(self):
+        assert self.is_element_present(*RegisterPageLocators.CUSTOMER_FORM), "Customer form is not present"
+
+    def add_first_name(self):
```
Comment (AramaisTyshchenko) [inline Tests/PageObject/RegisterPage.py:8]: make like this, first_name_input_field

File: Tests/PageObject/RegisterPage.py [hunk 18]
```diff
@@ -0,0 +1,72 @@
...
+        self.browser.find_element(*RegisterPageLocators.FIELD_USERNAME).send_keys(f"-1{first_name}1-")
+
+    def add_password(self):
+        self.browser.find_element(*RegisterPageLocators.FIELD_PASSWORD).send_keys(password)
+
+    def add_confirm(self):
+        self.browser.find_element(*RegisterPageLocators.FIELD_CONFIRM).send_keys(password)
+
+    def press_the_button_register(self):
+        self.click_on_button(*RegisterPageLocators.BUTTON_REGISTER)
```
Comment (AramaisTyshchenko) [inline Tests/PageObject/RegisterPage.py]: click should be inside test

File: Tests/Tests UI/test_base_page.py [hunk 19]
```diff
@@ -0,0 +1,25 @@
...
+    page_base.button_go_home()
+    page_base.open()
+    page_base.press_the_button_register_base()
+    page_base.button_go_contact()
+    page_base.open()
+    page_base.press_the_button_register_base()
+    page_base.button_go_aboutus()
+
+
+class TestUpMenu:
```
Comment (AramaisTyshchenko) [inline Tests/Tests UI/test_base_page.py]: please delete Class for test

File: Tests/Tests UI/test_base_page.py [hunk 20]
```diff
@@ -0,0 +1,25 @@
...
+    def test_guest_use_button_about_us(self, page_base):
+        page_base.button_go_about_us()
```
Comment (AramaisTyshchenko) [inline Tests/Tests UI/test_base_page.py]: please start the fixture once before all tests.

File: Tests/Tests UI/test_register_page.py [hunk 21]
```diff
@@ -0,0 +1,100 @@
+class TestPositive:
```
Comment (AramaisTyshchenko) [inline Tests/Tests UI/test_register_page.py]: remove Class

File: Tests/Tests UI/test_register_page.py [hunk 22]
```diff
@@ -0,0 +1,100 @@
...
+        page_register.button_go_home()
+        page_register.open()
+        page_register.press_the_button_go_to_register()
+        page_register.button_go_contact()
+        page_register.open()
+        page_register.press_the_button_go_to_register()
+        page_register.button_go_aboutus()
+
+
+class TestNegative:
```
Comment (AramaisTyshchenko) [inline Tests/Tests UI/test_register_page.py]: please remove Class

File: Tests/PageObject/HomePage.py [hunk 23]
```diff
@@ -0,0 +1,32 @@
+from Tests.PageObject.BasePage import BasePage
+from Tests.PageObject.locators import BasePageLocators, AboutUsPageLocators, ServicesPageLocators, \
+    ProductsAndLocationsPageLocators, AdminPagePageLocators
+
+
+class HomePage(BasePage):
+    def button_go_about_us(self):
+        self.click_on_button(*BasePageLocators.BUTTON_ABOUT_US)
```
Comment (AramaisTyshchenko) [inline Tests/PageObject/HomePage.py]: remove click

File: Tests/PageObject/HomePage.py [hunk 24]
```diff
@@ -0,0 +1,32 @@
...
+        assert self.is_element_present(*AboutUsPageLocators.LINK)
+
+    def button_go_services(self):
+        self.click_on_button(*BasePageLocators.BUTTON_SERVICES)
+        assert self.is_element_present(*ServicesPageLocators.HEADER1)
```
Comment (AramaisTyshchenko) [inline Tests/PageObject/HomePage.py]: remove verifications, verify only in test

File: Tests/PageObject/HomePage.py (hunk 23 above)
Comment (AramaisTyshchenko) [reply Tests/PageObject/HomePage.py]: or rename method to smth like go_to_about_us_page

File: Tests/Tests UI/test_home_page.py [hunk 25]
```diff
@@ -0,0 +1,44 @@
+def test_guest_use_buttons_gohome_contact_aboutus_from_register_page(page_home):
+    page_home.button_go_home.click()
+    assert page_home.should_be_image_go_home
```
Comment (AramaisTyshchenko) [inline Tests/Tests UI/test_home_page.py]: add comments to assert, if it fails

**Review by AramaisTyshchenko — APPROVED** [review]

**Pull request merged**